packages/@aws-cdk/aws-ec2/lib/connections.ts

```diff
@@ -118,60 +101,60 @@ export class Connections implements IConnections {
     public allowFromAnyIpv4(portRange: IPortRange, description: string) {
         this.allowFrom(new AnyIPv4(), portRange, description);
     }
-}
-
-/**
- * A class to orchestrate connections that already has default ports
- */
-export class DefaultConnections extends Connections {
-    public readonly defaultPortRange: IPortRange;
-
-    constructor(securityGroup: ISecurityGroup, defaultPortRangeProvider: IDefaultConnectable) {
-        // We take a IDefaultConnectable as an argument instead of the port
-        // range directly so (a) we force the containing construct to implement
-        // IDefaultConnectable and then (b) so they don't have to repeat the information.
-        //
-        // Slightly risky since this requires that the container initializes in the right order.
-        super(securityGroup);
-        this.defaultPortRange = defaultPortRangeProvider.defaultPortRange;
-
-        if (this.defaultPortRange == null) {
-            throw new Error("Ordering problem: create DefaultConnections() after initializing defaultPortRange");
-        }
-    }
 
     /**
      * Allow connections from the peer on our default port
      *
      * Even if the peer has a default port, we will always use our default port.
      */
     public allowDefaultPortFrom(other: IConnectable, description: string) {
+        if (!this.defaultPortRange) {
+            throw new Error('Cannot call allowDefaultPortFrom(): resource has no default port');
+        }
         this.allowFrom(other, this.defaultPortRange, description);
     }
 
     /**
      * Allow hosts inside the security group to connect to each other
      */
     public allowDefaultPortInternally(description: string) {
+        if (!this.defaultPortRange) {
+            throw new Error('Cannot call allowDefaultPortInternally(): resource has no default port');
+        }
         this.allowInternally(this.defaultPortRange, description);
     }
 
     /**
      * Allow default connections from all IPv4 ranges
      */
     public allowDefaultPortFromAnyIpv4(description: string) {
+        if (!this.defaultPortRange) {
+            throw new Error('Cannot call allowDefaultPortFromAnyIpv4(): resource has no default port');
+        }
         this.allowFromAnyIpv4(this.defaultPortRange, description);
     }
 }
 
+/**
+ * Connectable that represents a peer but doesn't modify any security groups
+ */
+class NullConnectable implements IConnectable {
+    public readonly connections: Connections;
+
+    constructor(connectionPeer: IConnectionPeer) {
+        this.connections = new SecurityGrouplessConnections(connectionPeer);
+    }
+}
+
 /**
  * This object is used by peers who don't allow reverse connections
  *
- * It still has an associated connection peer, but that peer does not
- * have any security groups to add connections to.
+ * Because Connections is no longer an interface but a concrete class,
```

Comment:
Nit: This sounds like a code review comment, not something that should be checked in.

Reply:
Kind of agree, but the next person coming along in this code base might want some content.

Do you prefer removing it? 

It doesn't actually need to be exported, so it becomes private documentation, I can also just do that.

Reply:
Ah that wasn't actually true. I would still like to retain it somewhere close to the code, but I agree it doesn't need to be a doc comment. Moved to implementation comment.